### PR TITLE
[PROGRAMMES-6901] Added schema for images

### PR DESCRIPTION
**Pull request by tomwinnington**
Ticket:
https://jira.dev.bbc.co.uk/browse/PROGRAMMES-6901

Link:
http://sandbox.bbc.co.uk/programmes/p03ts7dk/p03tqp9b

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

File: src/Controller/Helpers/SchemaHelper.php [hunk 1]
```diff
@@ -205,4 +208,29 @@ public function buildSchemaForContributor(Contribution $contribution): array
         }
         return $schema;
     }
+
+    public function buildSchemaForImage(Image $image, Programme $programme)
+    {
+        $schema = [
+            "@type" => "ImageObject",
+            "caption" => $programme->getTitle() . ' - ' . $image->getTitle() . ' - ' . $image->getLongestSynopsis(),
+            "thumbnail" => "https://ichef.bbci.co.uk/images/ic/224xn/" . $image->getPid() . "jpg",
```
Comment (Tasiobg) [inline src/Controller/Helpers/SchemaHelper.php]:
```suggestion
            "thumbnail" => "https://ichef.bbci.co.uk/images/ic/224xn/" . $image->getPid() . ".jpg",
```

File: src/Controller/Helpers/SchemaHelper.php [hunk 2]
```diff
@@ -205,4 +208,29 @@ public function buildSchemaForContributor(Contribution $contribution): array
...
+            "contentUrl" => $image->getUrl(560),
```
Comment (Tasiobg) [inline src/Controller/Helpers/SchemaHelper.php]:
```suggestion
            "contentUrl" => $image->getUrl(976),
```
I suggest to match the size of the image we display

File: src/Controller/Helpers/StructuredDataHelper.php [hunk 3]
```diff
@@ -97,6 +100,22 @@ public function getSchemaForNonActorContribution(Contribution $contribution): ar
         return $this->schemaHelper->buildSchemaForContributor($contribution);
     }
 
+
+    public function getSchemaForImage(Image $image, Programme $programme, Gallery $gallery, bool $showParent = true)
+    {
+        $schema = $this->schemaHelper->buildSchemaForImage($image, $programme, $gallery, $showParent);
```
Comment (Tasiobg) [inline src/Controller/Helpers/StructuredDataHelper.php]: PHPStan is throwing this error
```
Method App\Controller\Helpers\SchemaHelper::buildSchemaForImage()  
         invoked with 4 parameters, 2 required.
```

File: src/Controller/FindByPid/GalleryController.php [hunk 4]
```diff
@@ -62,4 +68,14 @@ public function getFirstImage(?string $imagePid, array $images): ?Image
         }
         return $image;
     }
+
+    public function getSchema(StructuredDataHelper $structuredDataHelper, Array $images, Programme $programme, Gallery $gallery)
```
Comment (Tasiobg) [inline src/Controller/FindByPid/GalleryController.php]:
```suggestion
    public function getSchema(StructuredDataHelper $structuredDataHelper, array $images, Programme $programme, Gallery $gallery)
```
I don't think this breaks anything but my understanding is that `array` should be lowercase

File: src/Controller/Helpers/SchemaHelper.php (hunk 2 above)
Comment (Tasiobg) [reply src/Controller/Helpers/SchemaHelper.php]: Not sure if there is any reason to keep it as 560 or just got missed when fixing something else.

File: src/Controller/FindByPid/GalleryController.php (hunk 4 above)
Comment (Tasiobg) [reply src/Controller/FindByPid/GalleryController.php]: I think this could have been missed

**Comment by Tasiobg** [comment]
test this please

**Pull request merged**